drivers/dai/nxp/micfil/micfil.c

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright 2025 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/drivers/dai.h>
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+
+#include "fsl_pdm.h"
+
+#define DT_DRV_COMPAT nxp_dai_micfil
+LOG_MODULE_REGISTER(nxp_dai_micfil);
+
+#define MICFIL_CLK_ROOT 24576000
+#define MICFIL_OSR_DEFAULT	16
+
+#define UINT_TO_MICFIL(x) ((PDM_Type *)(uintptr_t)(x))
+
+#define MICFIL_FIFO_BASE(inst) \
+	POINTER_TO_UINT(&(UINT_TO_MICFIL(DT_INST_REG_ADDR(inst))->DATACH[0]))
+
+#define MICFIL_DMA_HANDSHAKE(inst) \
+	((DT_INST_DMAS_CELL_BY_IDX(inst, 0, channel) & GENMASK(7, 0)) | \
+	((DT_INST_DMAS_CELL_BY_IDX(inst, 0, mux) << 8) & GENMASK(15, 8)))
+
+struct dai_nxp_micfil_data {
+	PDM_Type *base;
+	struct dai_config cfg;
+};
+
+struct dai_nxp_micfil_config {
+	const struct dai_properties *rx_props;
+	uint32_t fifo_depth;
```

Comment:
is this really needed? can't you just use `fifo_depth` from `rx_props`?

Reply:
dropped